src/kernel/scheduler.zig

```diff
@@ -348,14 +355,47 @@ fn rt_variable_preserved(allocator: *Allocator) void {
     log.info("SUCCESS: Scheduler variables preserved\n", .{});
 }
 
+fn rt_user_task(allocator: *Allocator, mem_profile: *const mem.MemProfile) void {
+    // 1. Create user VMM
+    var task_vmm = allocator.create(vmm.VirtualMemoryManager(arch.VmmPayload)) catch |e| {
+        panic(@errorReturnTrace(), "Failed to allocate user task VMM: {}\n", .{e});
+    };
+    task_vmm.* = vmm.VirtualMemoryManager(arch.VmmPayload).init(0, @ptrToInt(mem_profile.vaddr_start), allocator, arch.VMM_MAPPER, undefined) catch unreachable;
+    // 2. Create user task. The code will be loaded at address 0
+    var user_task = task.Task.create(0, false, task_vmm, allocator) catch |e| {
+        panic(@errorReturnTrace(), "Failed to create user task: {}\n", .{e});
+    };
+    // 3. Read the user program file from the filesystem
+    const user_program_file = fs.openFile("/user_program", .NO_CREATION) catch |e| {
+        panic(@errorReturnTrace(), "Failed to open /user_program: {}\n", .{e});
+    };
+    const code = user_program_file.read(1024) catch |e| {
+        panic(@errorReturnTrace(), "Failed to read user program file: {}\n", .{e});
+    };
+    defer allocator.free(code);
+    // 4. Allocate space in the vmm for the user_program
+    const code_start = task_vmm.alloc(std.mem.alignForward(code.len, vmm.BLOCK_SIZE) / vmm.BLOCK_SIZE, .{ .kernel = false, .writable = true, .cachable = true }) catch |e| {
+        panic(@errorReturnTrace(), "Failed to allocate VMM memory for user program code: {}\n", .{e});
+    } orelse unreachable;
+    if (code_start != 0) unreachable;
+    // 5. Copy user_program code over
+    vmm.kernel_vmm.copyDataToVMM(task_vmm, code, code_start) catch |e| {
+        panic(@errorReturnTrace(), "Failed to copy user code: {}\n", .{e});
+    };
+    // 6. Schedule it
+    scheduleTask(user_task, allocator) catch unreachable;
+    // 7. Profit!(?)
```

Comment:
Since the cpu state is arch-dependent, I've added a function to arch.zig that will check the saved state for the expected values. Let me know what you think of this.